drafts/arcalinea-tx-expiry/draft.rst

```diff
@@ -0,0 +1,75 @@
+**Title:** Transaction Expiry
+
+**Author:** Jay Graber
+
+**Status:** Active
+
+**Category:** Standards
+
+**Created:** 2018-01-09
+
+Abstract
+===========
+
+This is an Standards ZIP describing a new consensus rule to set an expiration time for transactions that are not mined to be removed from the mempool.
+
+Motivation
+===========
+
+Transactions that are too large or have insufficient fees are often not mined. This indeterminism is a source of confusion for users and wallets. Allowing transactions to set a block height after which it expires from the mempool would provide certainty around how long a transaction has to confirm before it is rejected by the network and must be re-sent.
+
+Advantages include improving performance by removing transactions unlikely to be mined, and potentially simplifying bidirectional payment channels by reducing the need to store and compress revocations for past states, since transactions not committed to the chain could expire and become invalid after a period of time.
+
+Specification
+===============
+
+Transactions will have a new field, nBlockExpiry, which will set the block height after which transactions will be removed from the mempool if they have not been mined.
+
+The data type for nBlockExpiry will be uint32_t, conforming to the structure of nLockTime. If used in combination with nLockTime, both nLockTime and nBlockExpiry must be block heights.
+
+For the example below, the last block that the transaction below could possibly be included in is 3539. After that, it will be removed from the mempool.
+
+````
+"txid": "17561b98cc77cd5a984bb959203e073b5f33cf14cbce90eb32b95ae2c796723f",
+"version": 3,
+"locktime": 2089,
+"blockexpiry": 3539,
+```
+
+Default: TBD. One suggestion is 576 blocks, or about 1 day assuming 2.5 minute block times. Can add a config option to set user's default.
+Minimum: No minimum
+Maximum: 500000000, about 380 years
+No limit: To set no limit on transactions (so that they do not expire), nBlockExpiry should be set to UINT_MAX.  
```

Comment:
UINT32_MAX or numeric_limits<uint32_t>::max() ... but since the code bases uses an int for the block height e.g. `(int)chainActive.Height()` this means that UINT_MAX as an int would represent a negative block height and thus cause many block height comparisons to fail.

Reply:
After conversation with Zooko on Rocketchat, and feedback from Linda, a proposal was made to change the default to 20 blocks.

Reply:
@daira what do you think the setting for "no expiry" should be?